File: room-manager/src/main/kotlin/org/rivero/roommanager/subscription/MealService.java
package org.rivero.roommanager.subscription;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Collection;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class MealService {
    private final MealCheckListRepository mealCheckListRepository;

    public void insert(MealCreateRequest mealCheckList) {
        mealCheckListRepository.insert(new MealCheckListDto(UUID.randomUUID().toString(), mealCheckList.month(), mealCheckList.checklist(), mealCheckList.consumerid(), mealCheckList.time()));
    }

    public Collection<MealCheckListDto> getList() {
        return mealCheckListRepository.getList();
    }

    public void updateOne(MealCreateRequest mealCheckList, String id) {
        mealCheckListRepository.updateOne(new MealCheckListDto(id, mealCheckList.month(), mealCheckList.checklist(), mealCheckList.consumerid(), mealCheckList.time()));
    }

    public MealCheckListDto getOne(String id) {
        return mealCheckListRepository.getOne(id);
    }
}

File: room-manager/src/main/kotlin/org/rivero/roommanager/subscription/MealService.kt
package org.rivero.roommanager.subscription

import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Service
import java.util.*

@Service
@RequiredArgsConstructor
class MealService {
    private val mealCheckListRepository: MealCheckListRepository? = null

    fun insert(mealCheckList: MealCreateRequest) {
        mealCheckListRepository!!.insert(
            MealCheckListDto(
                UUID.randomUUID().toString(),
                mealCheckList.month,
                mealCheckList.checklist,
                mealCheckList.consumerId,
                mealCheckList.time
            )
        )
    }

    val list: Collection<MealCheckListDto>
        get() = mealCheckListRepository!!.list

    fun updateOne(mealCheckList: MealCreateRequest, id: String) {
        mealCheckListRepository!!.updateOne(
            MealCheckListDto(
                id,
                mealCheckList.month,
                mealCheckList.checklist,
                mealCheckList.consumerId,
                mealCheckList.time
            )
        )
    }

    fun getOne(id: String): MealCheckListDto? {
        return mealCheckListRepository!!.getOne(id)
    }
}

